datanode/snapshot/current_snapshot.go

```diff
@@ -0,0 +1,145 @@
+package snapshot
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+)
+
+type CurrentStateSnapshot struct {
+	ChainID string
+	Height  int64
+}
+
+const currentStateSnapshotIdentifier = "currentstatesnapshot"
+
+func NewCurrentSnapshot(chainID string, height int64) CurrentStateSnapshot {
+	return CurrentStateSnapshot{
+		ChainID: chainID,
+		Height:  height,
+	}
+}
+
+func (s CurrentStateSnapshot) UncompressedDataDir() string {
+	return fmt.Sprintf("%s-%d-%s", s.ChainID, s.Height, currentStateSnapshotIdentifier)
+}
+
+func (s CurrentStateSnapshot) CompressedFileName() string {
+	return fmt.Sprintf("%s-%d-%s.tar.gz", s.ChainID, s.Height, currentStateSnapshotIdentifier)
+}
+
+func (s CurrentStateSnapshot) String() string {
+	return fmt.Sprintf("{Current State Snapshot Chain ID:%s Height:%d}", s.ChainID, s.Height)
+}
+
+func (s CurrentStateSnapshot) GetCopySQL(dbMetaData DatabaseMetadata, databaseSnapshotsPath string) []string {
+	var copySQL []string
+	for tableName, meta := range dbMetaData.TableNameToMetaData {
+		if !dbMetaData.TableNameToMetaData[tableName].Hypertable {
+			snapshotFile := filepath.Join(databaseSnapshotsPath, s.UncompressedDataDir(), tableName)
+			tableCopySQL := fmt.Sprintf(`copy (select * from %s order by %s) to '%s'`, tableName,
+				meta.SortOrder, snapshotFile)
+			copySQL = append(copySQL, tableCopySQL)
+		}
+	}
+
+	return copySQL
+}
+
+func FileExists(snapshotsDir string, file string) (bool, error) {
+	if _, err := os.Stat(filepath.Join(snapshotsDir, file)); err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+func GetCurrentStateSnapshots(snapshotsDir string) (string, map[int64]CurrentStateSnapshot, error) {
+	files, err := ioutil.ReadDir(snapshotsDir)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get files in snapshot directory:%w", err)
+	}
+
+	currentStateSnapshots := map[int64]CurrentStateSnapshot{}
+	chainID := ""
+	for _, file := range files {
+		if !file.IsDir() {
+			csSnapshot := fromCurrentStateSnapshotFileName(file.Name())
+
+			if csSnapshot == nil {
+				continue
+			}
+
+			if len(chainID) == 0 {
+				chainID = csSnapshot.ChainID
+			}
+
+			if csSnapshot.ChainID != chainID {
+				return "", nil, fmt.Errorf("current state snapshots for multiple chain ids exist in snapshots directory %s", snapshotsDir)
+			}
+
+			lockFileExists, err := FileExists(snapshotsDir, WriteLockFileName(csSnapshot.ChainID, csSnapshot.Height))
+			if err != nil {
+				return "", nil, fmt.Errorf("failed to check for lock file:%w", err)
+			}
+
+			if lockFileExists {
+				continue
+			}
+			currentStateSnapshots[csSnapshot.Height] = *csSnapshot
+		}
+	}
+
+	return chainID, currentStateSnapshots, nil
+}
+
+func fromCurrentStateSnapshotFileName(fileName string) *CurrentStateSnapshot {
+	re := regexp.MustCompile("(.*)-(\\d+)-" + currentStateSnapshotIdentifier + ".tar.gz")
```

Comment:
Must compile will panic if there's an error. It's nice when you want something like a global var I suppose, here, we may want to have some kind of error handling instead?

Reply:
this was fixed (and is still), what code do you see?

Reply:
Ah you're right, my grep picked up the one in `func fromHistoryFileName()` which should probably be changed too.

Reply:
sorted that one as well